addons/mail/static/src/model/model_core.js

```diff
@@ -456,3 +318,146 @@ export function registerModel({ fields, identifyingMode = 'and', lifecycleHooks,
         addOnChanges(name, onChanges);
     }
 }
+
+export function registerPatch({ fields, lifecycleHooks, modelMethods, name, onChanges, recordMethods }) {
+    if (!name) {
+        throw new Error("Patch is lacking the name of the model to be patched.");
+    }
+    const allowedSectionNames = ['name', 'lifecycleHooks', 'modelMethods', 'recordMethods', 'fields', 'onChanges'];
+    const invalidSectionNames = Object.keys(arguments['0']).filter(x => !allowedSectionNames.includes(x));
```

Comment:
In this case I think we would name the parameter and deconstruct inside the method instead of using `arguments`